File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/ValidValueValidator.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.base.comum.validadores.CustomValidator;
import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidRule;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidValue;
import java.util.HashMap;
import java.util.Map;

public class ValidValueValidator
extends CustomValidator<ValidValue> {
    private ValidRule validRule;
    private Map<String, Object> parameters = new HashMap<String, Object>();

    @Override
    public Object getParameter(String key) {
        if (this.getAnnotationParameters().length > 0) {
            for (int i = 0; i < this.getAnnotationParameters().length; ++i) {
                if (!Integer.valueOf(i).toString().equals(key.trim())) continue;
                return this.getAnnotationParameters()[i];
            }
        } else {
            return this.getParameters().get(key);
        }
        return null;
    }

    @Override
    public String getMessage() {
        if (Utils.naoNulo(this.validRule)) {
            return this.validRule.getMessage().name();
        }
        return ((ValidValue)this.getAnnotation()).message();
    }

    @Override
    public byte[] getGroups() {
        return ((ValidValue)this.getAnnotation()).groups();
    }

    @Override
    protected String getCondition() {
        return ((ValidValue)this.getAnnotation()).condition();
    }

    public String[] getAnnotationParameters() {
        return ((ValidValue)this.getAnnotation()).parameters();
    }

    public int getFlag() {
        return ((ValidValue)this.getAnnotation()).flag();
    }

    public Map<String, Object> getParameters() {
        return this.parameters;
    }

    @Override
    public boolean isValid(ValidatorContext context) {
        try {
            this.validRule = ((ValidValue)this.getAnnotation()).validRule().newInstance();
            return this.validRule.isValid(this, context);
        }
        catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }
}

File: src/lib/pjecalc/core/comum/validators/valid-value-validator.ts
/**
 * PJe-Calc v2.15.1 — ValidValueValidator
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.ValidValueValidator
 */
import { CustomValidator } from './custom-validator';
import type { ValidRule } from './valid-rule';
import type { ValidValue_ } from './valid-value';
import type { ValidatorContext } from './validator-context';
import { logger } from '@/lib/logger';

export class ValidValueValidator extends CustomValidator<ValidValue_> {
  private validRule: ValidRule | null = null;
  private parametersMap: Map<string, unknown> = new Map();

  getMessage(): string {
    if (this.validRule != null) {
      const msg = this.validRule.getMessage();
      return msg != null ? String(msg) : this.spec.message;
    }
    return this.spec.message;
  }

  getGroups(): number[] { return this.spec.groups; }
  protected getCondition(): string { return this.spec.condition; }
  getFlag(): number { return this.spec.flag; }

  getAnnotationParameters(): string[] { return this.spec.parameters; }
  getParameters(): Map<string, unknown> { return this.parametersMap; }

  getParameter(key: string): unknown {
    const annotationParams = this.getAnnotationParameters();
    if (annotationParams.length > 0) {
      for (let i = 0; i < annotationParams.length; i++) {
        if (String(i) === key.trim()) return annotationParams[i];
      }
      return null;
    }
    return this.parametersMap.get(key);
  }

  isValid(context: ValidatorContext): boolean {
    try {
      this.validRule = new this.spec.validRule();
      return this.validRule.isValid(this, context);
    } catch (e) {
      // eslint-disable-next-line no-console
      logger.error(e)
      return false;
    }
  }
}
